File: HB/pysrc/프로그래머스/레벨0/Day20/다항식_더하기.py
def solution(polynomial):
    answer = ''
    lst = polynomial.split(" ")
    x_cnt = 0
    cnt = 0
    for i in range(len(lst)):
        if "x" in lst[i]:
            if lst[i] == "x":
                x_cnt += 1
            else:
                x_cnt += int(lst[i][0:-2])
        elif lst[i] != '+':
            cnt += int(lst[i])
    if x_cnt > 0:
        if x_cnt == 1:
            answer += 'x'
        else:
            answer += str(x_cnt)+'x'
    if cnt >= 0 and x_cnt > 0:
        answer += ' + '+str(cnt)
    elif cnt >= 0 and x_cnt == 0:
        answer = str(cnt)
    print(answer)
    return answer
solution("3 + x")

File: SB/src/프로그래머스/레벨0/Day19/다항식_더하기.java
package 프로그래머스.레벨0.Day19;

public class 다항식_더하기 {
    class Solution {
        public String solution(String polynomial) {
            String answer = "";
            String sArr[] = polynomial.split(" ");
            int strNum = 0;
            int num = 0;

            for (int i = 0; i < sArr.length; i++) {
                if (sArr[i].contains("x") && sArr[i].length() > 1) {
                    strNum += Integer.parseInt(sArr[i].substring(0, sArr[i].length() -1));
                } else if (sArr[i].contains("x") && sArr[i].length() == 1) {
                    strNum += 1;
                } else if (!sArr[i].contains("x") && !sArr[i].contains("+")) {
                    num += Integer.parseInt(sArr[i]);
                }
            }

            if (strNum > 0 && num > 0) {
                if (strNum == 1) {
                    answer = "x" + " + " + String.valueOf(num);
                } else {
                    answer = String.valueOf(strNum) + "x" + " + " + String.valueOf(num);
                }

            } else if (strNum > 0) {
                if (strNum == 1) {
                    answer = "x";
                } else {
                    answer = String.valueOf(strNum) + "x";
                }

            } else if (num > 0) {
                answer = String.valueOf(num);
            }

            return answer;
        }
    }
}
